File: cpp_files/raylib/single_file_programs/center_and_draw_text.cpp
#include "center_and_draw_text.h"

#include <cassert>

void center_and_draw_text(const char *text, Rectangle layout_rect, Font font,
                          float font_size, float spacing, Color color)
{
    assert(text);

    Vector2 text_size = MeasureTextEx(font, text, font_size, spacing);
    Vector2 pos       = (Vector2){
        layout_rect.x + layout_rect.width / 2.0f - text_size.x / 2.0f,
        layout_rect.y + layout_rect.height / 2.0f - text_size.y / 2.0f,
    };

    DrawTextEx(font, text, pos, font_size, spacing, color);
}

File: python_files/raylib/single_file_programs/center_and_draw_text.py
import pyray as p


def center_and_draw_text(
    text: str,
    layout_rect: p.Rectangle,
    font_size: float = 30.0,
    spacing: float = 2.0,
    color: p.Color = p.DARKGRAY,
):
    font: p.Font = p.get_font_default()
    text_size: p.Vector2 = p.measure_text_ex(font, text, font_size, spacing)
    pos: p.Vector2 = p.Vector2(
        layout_rect.x + layout_rect.width / 2.0 - text_size.x / 2.0,
        layout_rect.y + layout_rect.height / 2.0 - text_size.y / 2.0,
    )

    p.draw_text_ex(font, text, pos, font_size, spacing, color)
